os/board/rtl8730e/src/component/os/tizenrt/rtk_wifi_csi.c

```diff
@@ -0,0 +1,261 @@
+/****************************************************************************
+ *
+ * Copyright 2024 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <poll.h>
+#include <semaphore.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <tinyara/kmalloc.h>
+#include "rtk_wifi_csi.h"
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+static int rtk_wifi_csi_ioctl(FAR struct wifi_csi_lowerhalf_s *dev, int cmd, unsigned long arg);
+static int rtk_wifi_csi_getmacaddr(FAR struct wifi_csi_lowerhalf_s *dev, char *mac_addr);
+static int rtk_wifi_csi_getcsidata(FAR struct wifi_csi_lowerhalf_s *dev, char *buffer, size_t buflen);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+static const struct wifi_csi_ops_s g_wificsiops = {
+	NULL,                           /* read           */
+	rtk_wifi_csi_ioctl,             /* ioctl          */
+	rtk_wifi_csi_getmacaddr,        /* getmacaddr     */
+	rtk_wifi_csi_getcsidata,        /* getcsidata     */
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static void data_ready_listener(char *buf, int buf_len, int flags, void *userdata)
+{
+	struct wifi_csi_lowerhalf_s *dev = (struct wifi_csi_lowerhalf_s *)userdata;
+	FAR struct wifi_csi_dev_s *priv = (FAR struct wifi_csi_dev_s *)dev;
+	// DEBUGASSERT(priv && priv->dev.upper && apb); //I think we should use this too.
+	if (!dev) {
+		csidbg("ERROR: dev null\n");
+		return;
+	}
+	if (!priv) {
+		csidbg("ERROR: priv null\n");
+		return;
+	}
```

Comment:
Is there any case which dev is NOT null and priv is null? I mean that checking priv is necessary?